mscore/scoretreeview.cpp

```diff
@@ -0,0 +1,100 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#include "scoretreeview.h"
+
+namespace Ms {
+
+//---------------------------------------------------------
+//   selectElementAndChildren
+//---------------------------------------------------------
+
+static void selectElementAndChildren(Element* el)
+{
+    el->score()->selection().add(el);
+    for (ScoreElement* child : (*el)) {
+        if (child) {
+            selectElementAndChildren(toElement(child));
+        }
+    }
+}
+
+//---------------------------------------------------------
+//   ScoreTreeView
+//---------------------------------------------------------
+
+ScoreTreeView::ScoreTreeView(QWidget* parent)
+    : QTreeView(parent)
+{
+}
+
+//---------------------------------------------------------
+//   ~ScoreTreeView
+//---------------------------------------------------------
+
+ScoreTreeView::~ScoreTreeView()
+{
+}
+
+//---------------------------------------------------------
+//   setModel
+//---------------------------------------------------------
+
+void ScoreTreeView::setModel(QAbstractItemModel* model)
+{
+    Q_ASSERT(model);  // ensure there is a model
+    Q_ASSERT(dynamic_cast<ScoreItemModel*>(model));  // ensure model is correct type
+    QTreeView::setModel(model);
+}
+
+//---------------------------------------------------------
+//   model
+//---------------------------------------------------------
+
+ScoreItemModel* ScoreTreeView::scoreItemModel() const
+{
+    return static_cast<ScoreItemModel*>(QTreeView::model());
+}
+
+//---------------------------------------------------------
+//   currentChanged
+//---------------------------------------------------------
+
+void ScoreTreeView::currentChanged(const QModelIndex& current, const QModelIndex& previous)
+{
+    selectElementAtIndex(current);
+    QTreeView::currentChanged(current, previous);
+}
+
+//---------------------------------------------------------
+//   selectElementAtIndex
```

Comment:
We need a doc-style comment here to explain that this function selects elements in the score (and therefore in the ScoreView) rather than elements in the tree view.